.travis.yml

```diff
@@ -23,9 +27,26 @@ env:
   - CONFIG=ruby22
   - CONFIG=jruby
 matrix:
-  allow_failures:
+  exclude:
+    # It's nontrivial to programmatically install a new JDK from the command
+    # line on OS X, so we rely on testing on Linux for Java code.
+    - os: osx
+      env: CONFIG=java_jdk6
+    - os: osx
+      env: CONFIG=java_jdk7
+    - os: osx
+      env: CONFIG=java_oracle7
+    - os: osx
+      env: CONFIG=javanano_jdk6
+    - os: osx
+      env: CONFIG=javanano_jdk7
+    - os: osx
+      env: CONFIG=javanano_oracle7
+    # Requires installing mono, currently travis.sh is doing that with apt-get
+    # which doesn't work on OS X.
     - os: osx
       env: CONFIG=csharp
+    # Does ruby install via ruby/travis-test.sh, currently doesn't work on OS X.
     - os: osx
```

Comment:
I think we should still mark these two ruby tests as "allow_failures". They are expected to work and @haberman is working on fixing them.